gneiss/regression/_ols.py

```diff
@@ -298,3 +318,144 @@ def r2(self):
 
         sst = sse + ssr
         return 1 - sse / sst
+
+    @property
+    def mse(self):
+        sse = sum([r.ssr for r in self.results])
+        dfe = self.results[0].df_resid
+        return sse / dfe
+
+    def loo(self, regularized=False, **kwargs):
+        """ Leave one out cross-validation.
+
+        Calculates summary statistics for each iteraction of
+        leave one out cross-validation, specially `mse` on entire model
+        and `pred_err` to measure prediction error.
+
+        Parameters
+        ----------
+        regularized : bool
+            Specifies if a regularization procedure should be used
+            when performing the fit.
+        **kwargs : dict
+           Keyword arguments used to tune the parameter estimation.
+
+        Returns
+        -------
+        pd.DataFrame
+           mse : np.array, float
+               Mean sum of squares error for each iteration of
+               the cross validation.
+           pred_err : np.array, float
+               Prediction mean sum of squares error for each iteration of
+               the cross validation.
+
+        See Also
+        --------
+        fit
+        """
+        params = {'fit_regularized': False}
+        for key in params:
+            params[key] = kwargs.get(key, params[key])
+            kwargs.pop(key, None)
+
+        nobs = self.balances.shape[0]  # number of observations (i.e. samples)
+        cv_iter = LeaveOneOut(nobs)
+        endog = self.balances
+        exog_names = self.results[0].model.exog_names
+        exog = pd.DataFrame(self.results[0].model.exog,
+                            index=self.balances.index,
+                            columns=exog_names)
+        results = pd.DataFrame(index=self.balances.index,
+                               columns=['mse', 'pred_err'])
+        i = 0
```

Comment:
I think there are 2 more pythonic way of doing this is, I'm unsure which one is the best for your case. Since `i` is just an index, this should work:
```python
for sample_id, (inidx, outidx) in zip(self.balances.index, cv_iter):
   etc...
```

However, If you still need `i` for something else that I may be missing, then this is the pythonic way of doing the index counting:

```python
for i, (inidx, outidx) in enumerate(cv_iter):
    ....
```
And remove the `i+=1` from the end of the for loop.